io_scene_vox.py

```diff
@@ -203,53 +220,147 @@ def import_vox(path, *, voxel_spacing=1, voxel_size=1,
+
+    print("Linked voxels:", len(to_link))
 
-    bpy.context.scene.update()
+    layer = bpy.context.view_layer
+    layer.update()
 
     print('\nSuccessfully imported {} in {:.3f} sec'.format(path, time.time() - time_start))
     return {'FINISHED'}
 
 
+#
+#==================================================================================================
+#     Change Diffuse shader to Emission shader without affecting shader color
+#     http://web.purplefrog.com/~thoth/blender/python-cookbook/change-diffuse-to-emission-node.html 
+#==================================================================================================
+#
+import bpy
+
+def replace_with_emission(node, node_tree):
+    new_node = node_tree.nodes.new('ShaderNodeEmission')
+    connected_sockets_out = []
+    sock = node.inputs[0]
+    if len(sock.links)>0:
+        color_link = sock.links[0].from_socket
+    else:
+        color_link=None
+    defaults_in = sock.default_value[:]
+
+    for sock in node.outputs:
+        if len(sock.links)>0:
+            connected_sockets_out.append( sock.links[0].to_socket)
+        else:
+            connected_sockets_out.append(None)
+
+    #print( defaults_in )
+
+    new_node.location = (node.location.x, node.location.y)
+
+    if color_link is not None:
+        node_tree.links.new(new_node.inputs[0], color_link)
+    new_node.inputs[0].default_value = defaults_in
+
+    if connected_sockets_out[0] is not None:
+        node_tree.links.new(connected_sockets_out[0], new_node.outputs[0])
+
+
+def material_diffuse_to_emission(mat):
+
+    doomed=[]
+    for node in mat.node_tree.nodes:
+        if node.type=='BSDF_DIFFUSE' or node.type=='BSDF_PRINCIPLED':
+            replace_with_emission(node, mat.node_tree)
+            doomed.append(node)
+        else:
+            print(node.type)
+
+    # wait until we are done iterating and adding before we start wrecking things
+    for node in doomed:
+        mat.node_tree.nodes.remove(node)
+
+
+def replace_on_selected_objects():
+    mats = set()
+    for obj in bpy.context.scene.objects:
+        if obj.select_set:
+            for slot in obj.material_slots:
+                mats.add(slot.material)
+
+    for mat in mats:
+        material_diffuse_to_emission(mat)
+
+def replace_in_all_materials():
+    for mat in bpy.data.materials:
+        material_diffuse_to_emission(mat)
+
+
+#
+#==================================================================================================
+#     Register - Unregister - MAIN
+#==================================================================================================
+#
+__classes__ = (
+    ImportVOX,
+)
+
+
 def menu_func_import(self, context):
     self.layout.operator(ImportVOX.bl_idname, text="MagicaVoxel (.vox)")
 
 
 def register():
-    bpy.utils.register_module(__name__)
-    bpy.types.INFO_MT_file_import.append(menu_func_import)
+    from bpy.utils import register_class
```

Comment:
import statements at start of file, please

Reply:
menu_func_import, register, unregister and the "main" calling register()?

Should not be at the end of file as they need other classes implemented? Excuse me but I never programmed in Python before.

Reply:
Apologies for lack of clarity.

`from bpy.utils import register_class`  is just importing this `register_class` function for us to use here.
Generally I prefer all the import statements to be collected at the top of the file.

Although, in this case, we could also replace the line `register_class(cls)` with `bpy.utils.register_class(cls)` and skip the import entirely (as we import the base `bpy` right at the start)